### PR TITLE
Get-Spn to better align with setspn -l

### LINKED ISSUE #713: BUG: Get-DbaSpn
### Is this a *feature* OR *bug*:
bug

Latest version release of dbatools

result of Get-DbaSpn

> ComputerName : Myserver.domain.net
InstanceName : MSSQLSERVER
SqlProduct : SQL Server 20xx Enterprise Edition (64-bit)
InstanceServiceAccount : NT SERVICE\MSSQLSERVER
RequiredSPN: MSSQLSvc/Myserver.domain.net:port
IsSet : False
Cluster : False
TcpEnabled : True
Port : MyPort
DynamicPort : False
Warning : None
Error : SPN missing

However with setspn -l Myserver:

> MSSQLSvc/Myserver.domain.net:Port
MSSQLSvc/Myserver.domain.net
Since my SQL is running at [NT SERVICE\MSSQLSERVER], Get-DbaSpn does not show a valid SPN but the old school setspn -l does show a valid SPN.

Looks like something that just needs to be investigated a little bit further. Either way...AWESOME WORK.

I made comment of this same issue on: https://github.com/sqlcollaborative/dbatools/issues/685. I think you could probably close this issue if you address what I found in https://github.com/sqlcollaborative/dbatools/issues/685

thanks

### PR DESCRIPTION
**Pull request by ctrlbold**
Any suggestions on the new parameter name? 
fixes #713 and #685

### CODE REVIEW

**Comment by ctrlbold** [comment]
What I did was set ComputerName to $env:COMPUTERNAME by default so if only an $accountname was specified, it'd be $env:COMPUTERNAME + $accountname, like setspn. 

This effectively removed the functionality to search for all SPNs set by $accountname so I added a new switch (crappy name, feel free to suggest alts) named $ShowAccountNameAll that will show all SPNs registered to that $accountname on the domain.

File: functions/Get-DbaSpn.ps1 [hunk 1]
```diff
@@ -71,76 +68,81 @@ Returns a custom object with SearchTerm (ServerName) and the SPNs that were foun
 			}
 		}
 		
-		ForEach ($account in $AccountName)
+		if ($ShowAccountNameAll)
```
Comment (FriedrichWeinmann) [inline functions/Get-DbaSpn.ps1]: Please use regions to split different sections of your code.
When using PSS:
Mark all of one logical section.
Press "CTRL" + "R"
Give it an acceptable name.
Makes it easier to read and maintain

File: functions/Get-DbaSpn.ps1 [hunk 2]
```diff
@@ -71,76 +68,81 @@ Returns a custom object with SearchTerm (ServerName) and the SPNs that were foun
...
-					ServiceClass = "MSSQLSvc" # $serviceclass
-					Port = $port
-					SPN = $spn
-                }
 			}
+			continue
 		}
 		
-		
 		foreach ($server in $ComputerName)
```
Comment (FriedrichWeinmann) [inline functions/Get-DbaSpn.ps1]: It would be customary to use $Computer, rather than $Server here.

**Comment by FriedrichWeinmann** [comment]
Set it to approved, since I'm going to bed now - you'll know yourself well enough when it's as done as it gets :)

File: functions/Get-DbaSpn.ps1 [hunk 3]
```diff
@@ -48,20 +50,15 @@ Returns a custom object with SearchTerm (ServerName) and the SPNs that were foun
     [cmdletbinding()]
 	param (
         [Parameter(Mandatory = $false,ValueFromPipeline = $true)]
-        [string[]]$ComputerName,
+        [string[]]$ComputerName = $env:COMPUTERNAME,
         [Parameter(Mandatory = $false)]
 		[string[]]$AccountName,
+		[switch]$ShowAccountNameAll,
```
Comment (ctrlbold) [inline functions/Get-DbaSpn.ps1]: I like ByAccount, thanks! 

The parametersets would work to keep ByAccount and AccountName together - but there are definitely times when users would want to check a remote host with another AccountName. Brain isn't working, pass for now.

File: functions/Get-DbaSpn.ps1 (hunk 1 above)
Comment (ctrlbold) [inline functions/Get-DbaSpn.ps1]: 1.0 ;)

File: functions/Get-DbaSpn.ps1 (hunk 2 above)
Comment (ctrlbold) [inline functions/Get-DbaSpn.ps1]: done!

File: functions/Get-DbaSpn.ps1 [hunk 4]
```diff
@@ -48,20 +50,15 @@ Returns a custom object with SearchTerm (ServerName) and the SPNs that were foun
...
 	process
-	{
-		if (!$ComputerName -and !$AccountName)
-		{
-			$ComputerName = $env:computername
-			$AccountName = "$env:USERDOMAIN\$env:USERNAME"
-		}
-		
+	{	
 		if ($ComputerName)
```
Comment (ctrlbold) [inline functions/Get-DbaSpn.ps1]: true, i had to move it a bunch of stuff to ensure this still worked, please recheck my logic.

@drewfurgiuele - please check this PR and approve. I changed the functionality.

File: functions/Get-DbaSpn.ps1 (hunk 1 above)
Comment (wsmelton) [inline functions/Get-DbaSpn.ps1]: Regions are not that useful at seperating logic in your code unless the editor that you use supports it. Which not every editor used by contributors is going to support it, so it is not really any different than just adding comments.

File: functions/Get-DbaSpn.ps1 [hunk 5]
```diff
@@ -71,76 +68,81 @@ Returns a custom object with SearchTerm (ServerName) and the SPNs that were foun
...
+				
+				Write-Verbose "Looking for account $account..."
+				
+				try
+				{
+					$Result = $adsearch.FindOne()
+				}
+				catch
+				{
+					Write-Warning "AD lookup failure. This may be because the hostname ($computer) was not resolvable within the domain ($domain) or the SQL Server service account ($serviceaccount) couldn't be found in domain."
```
Comment (ctrlbold) [inline functions/Get-DbaSpn.ps1]: Can you give the syntax plz? LOL :/. Then I'll remember it.

**Comment by ctrlbold** [comment]
Will add Stop-Process in 1.1

**Pull request merged**